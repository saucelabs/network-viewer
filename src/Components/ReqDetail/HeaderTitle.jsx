import React from 'react';
import PropTypes from 'prop-types';

import { HEADERS_TITLES } from './../../constants';
import IconCaretUp from './../../icons/IconCaretUp';
import Styles from './Headers.styles.scss';

const HeaderTitle = ({ onClick, eventKey }) => (
  <div
    className={Styles['header-title']}
    onClick={() => onClick(HEADERS_TITLES[eventKey].key)}
    role="button"
    tabIndex={0}
  >
    <IconCaretUp className={Styles['caret-icon']} />
    {HEADERS_TITLES[eventKey].name}
  </div>
);

HeaderTitle.propTypes = {
  eventKey: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
};

export default HeaderTitle;
