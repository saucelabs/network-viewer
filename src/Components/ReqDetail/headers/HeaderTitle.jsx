import React from 'react';
import PropTypes from 'prop-types';

import { HEADERS_TITLES, PAYLOAD_CAPTIONS } from './../../../constants';
import IconCaretUp from './../../../icons/IconCaretUp';
import Styles from './../Headers.styles.scss';

const HeaderTitle = ({
  onClick,
  eventKey,
  isEncodeEnabled,
  onPayloadTransform,
  isPayloadTransformed,
  isParseEnabled,
}) => {
  const payloadStatus = PAYLOAD_CAPTIONS[isParseEnabled ? 'parse' : 'encode'][isPayloadTransformed];

  return (
    <div className={Styles['header-title']}>
      <span
        onClick={() => onClick(HEADERS_TITLES[eventKey].key)}
        role="button"
        tabIndex={0}
      >
        <IconCaretUp className={Styles['caret-icon']} />
        {HEADERS_TITLES[eventKey].name}
      </span>
      {(isEncodeEnabled || isParseEnabled) && (
        <span
          className={Styles['encode-url']}
          onClick={onPayloadTransform}
          role="button"
          tabIndex={0}
        >
          {`view ${payloadStatus}`}
        </span>
      )}
    </div>
  );
};

HeaderTitle.propTypes = {
  eventKey: PropTypes.string.isRequired,
  isEncodeEnabled: PropTypes.bool,
  isParseEnabled: PropTypes.bool,
  isPayloadTransformed: PropTypes.bool,
  onClick: PropTypes.func.isRequired,
  onPayloadTransform: PropTypes.func.isRequired,
};

HeaderTitle.defaultProps = {
  isEncodeEnabled: false,
  isParseEnabled: false,
  isPayloadTransformed: true,
};

export default HeaderTitle;
