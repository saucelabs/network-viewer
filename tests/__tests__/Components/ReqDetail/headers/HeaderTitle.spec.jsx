import React from 'react';
import { shallow } from 'enzyme';

import HeaderTitle from './../../../../../src/Components/ReqDetail/headers/HeaderTitle';

describe('HeaderTitle', () => {
  const props = {
    eventKey: 'general',
    onClick: () => {},
    onChangeEncode: () => {},
  };

  it('renders without crashing', () => {
    const element = shallow(<HeaderTitle {...props} />);
    expect(element).toMatchSnapshot();
  });
});
