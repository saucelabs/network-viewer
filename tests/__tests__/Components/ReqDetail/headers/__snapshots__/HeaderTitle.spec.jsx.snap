// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderTitle renders without crashing 1`] = `
<div>
  <span
    onClick={[Function]}
    role="button"
    tabIndex={0}
  >
    <IconCaretUp
      className=""
    />
    General
  </span>
</div>
`;
